tests/Provider/BaseProviderThisVerOrBelowTest.php

```diff
@@ -0,0 +1,210 @@
+
+class BaseProviderThisVerOrBelowTest extends AbstractProviderTest
+{
+    public function getProvider()
+    {
+        $adapter = $this->createMock(Adapter::class);
+
+        $filesystem = $this->getMockBuilder(Filesystem::class)
+            ->setMethods(['get'])
+            ->setConstructorArgs([$adapter])
+            ->getMock();
+        $file = $this->getMockBuilder(File::class)
+            ->setConstructorArgs(['foo', $filesystem])
+            ->getMock();
+
+        $cdn = new Server('/uploads/media');
+
+        $generator = new DefaultGenerator();
+
+        $thumbnail = $this->createMock(ThumbnailInterface::class);
+
+        $metadata = $this->createMock(MetadataBuilderInterface::class);
+
+        $provider = new TestBaseProvider('test', $filesystem, $cdn, $generator, $thumbnail, $metadata);
+        $this->assertInstanceOf(BaseProvider::class, $provider);
+
+        return $provider;
+    }
+
+    public function testPostRemove(): void
+    {
+        $provider = $this->getProvider();
+        $media = new Media();
+        $media->setId(1399);
+        $media->setProviderReference('1f981a048e7d8b671415d17e9633abc0059df394.png');
+        
+        $provider->preRemove($media);
+        
+        // Emulate an object detached from the EntityManager.
+        $media->setId(null);
+        $provider->postRemove($media);
+    }
+}
+
+class TestBaseProvider extends BaseProvider
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getHelperProperties(MediaInterface $media, $format, $options = []): void
+    {
+        // TODO: Implement getHelperProperties() method.
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function postPersist(MediaInterface $media): void
+    {
+        // TODO: Implement postPersist() method.
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildEditForm(FormMapper $form): void
+    {
+        // TODO: Implement buildEditForm() method.
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildCreateForm(FormMapper $form): void
+    {
+        // TODO: Implement buildCreateForm() method.
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function postUpdate(MediaInterface $media): void
+    {
+        // TODO: Implement postUpdate() method.
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAbsolutePath(MediaInterface $media): void
+    {
+        // TODO: Implement getAbsolutePath() method.
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getReferenceImage(MediaInterface $media): string
+    {
+        // A copy of the code from \Sonata\MediaBundle\Provider\FileProvider::getReferenceImage()
+        $path = sprintf('%s/%s',
+            $this->generatePath($media),
+            $media->getProviderReference()
+        );
+        
+        // Test code.
+        AbstractProviderTest::assertSame('/0001/02/1f981a048e7d8b671415d17e9633abc0059df394.png', $path);
```

Comment:
`AbstractProviderTest` => `self`.
BTW, fixture methods should avoid the testing responsibility. The expected path should not be hardcoded, it should be provided by context (argument or so).